lib/cli/src/commands/create_exe.rs

```diff
@@ -419,32 +418,102 @@ impl CreateExe {
+                        files.iter().find(|f| f.ends_with("libwasmer.a")).cloned().ok_or_else(|| {
+                            anyhow!("Could not find libwasmer.a for {} target in the provided tarball path (files = {files:#?})", target)})?
                     } else {
-                        #[cfg(feature = "http")]
-                        {
-                            let release = http_fetch::get_latest_release()?;
-                            let tarball = http_fetch::download_release(release, target.clone())?;
-                            let target_file_path = tarball
+                        // check if the tarball for the target already exists locally
+                        let local_tarball = std::fs::read_dir(get_libwasmer_cache_path()?)?
+                            .filter_map(|e| e.ok())
+                            .filter_map(|e| {
+                                let path = format!("{}", e.path().display());
+                                if path.ends_with(".tar.gz") {
+                                    Some(e.path())
+                                } else {
+                                    None
+                                }
+                            })
+                            .filter_map(|p| {
+                                if let Architecture::Aarch64(_) = target.architecture {
+                                    if !p.file_name()?.to_str()?.contains("aarch64") {
+                                        return None;
+                                    }
+                                }
+
+                                if let Architecture::X86_64 = target.architecture {
+                                    if !p.file_name()?.to_str()?.contains("x86_64") {
+                                        return None;
+                                    }
+                                }
+
+                                if let OperatingSystem::Windows = target.operating_system {
+                                    if !p.file_name()?.to_str()?.contains("windows") {
+                                        return None;
+                                    }
+                                }
+
+                                if let OperatingSystem::Darwin = target.operating_system {
+                                    if !(p.file_name()?.to_str()?.contains("apple")
+                                        || p.file_name()?.to_str()?.contains("darwin"))
+                                    {
+                                        return None;
+                                    }
+                                }
+
+                                if let OperatingSystem::Linux = target.operating_system {
+                                    if !p.file_name()?.to_str()?.contains("linux") {
+                                        return None;
+                                    }
+                                }
+
+                                Some(p)
```

Comment:
Can you pull this out into its own helper function?

Reply:
done